include/onnxruntime/core/framework/allocator.h

```diff
@@ -24,6 +24,8 @@ constexpr const char* CUDA = "Cuda";
 constexpr const char* CUDA_PINNED = "CudaPinned";
 constexpr const char* MIGRAPHX = "MIGraphX";
 constexpr const char* MIGRAPHX_PINNED = "MIGraphXPinned";
+constexpr const char* ROCM = "Rocm";
+constexpr const char* ROCM_PINNED = "RocmPinned";
```

Comment:
In current design, I intended to use CUDA_PINNED and CUDA for ROCm EP as well. The idea is that they will be renamed to a common names, e.g. GPU and GPU_PINNED, so more code could be shared between ROCm EP and CUDA EP. Otherwise, we need to always duplicate this kind of definitions. That being said, let's change the changes below related to ROCM and ROCM_PINNED back to the original one.